enhancements/kueue/kueue.md

```diff
@@ -0,0 +1,539 @@
+	// ManagedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+	// FairSharing controls the fair sharing semantics across the cluster.
+	FairSharing *configapi.FairSharing `json:"fairSharing,omitempty"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+```
+
+We will use the kueue configuration [API](https://github.com/kubernetes-sigs/kueue/blob/main/apis/config/v1beta1/configuration_types.go).
+
+The operator reads the kueue configuration and generates a ConfigMap that the kueue manager deployment will use. 
+We will only expose a limited amount of configurations.
+
+The flow is as follows:
+
+1) User specifies a Kueue CRD
+1) Configuration is specified
+1) A Configmap will be created with the configuration specified
+1) Kueue deployment will use that configmap in the operand deployment.
+
+The operator will manage these components necessary for Kueue. 
+- Custom Resources
+- Cluster Roles
+- Cluster Role Bindings
+- Services
+- Mutating Webhook
+- Validating Webhook
+- ServiceMonitors
+- APIService for visibility
+- Roles
+- Certificates
+
+The operator will create and apply these resources to the cluster.
+
+
+### Workflow Description
+
+#### Batch Job Administrator
+
+As a admin, I want to enable Kueue to manage batch jobs.
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+   config:
+     integrations:
+       frameworks:
+         - "batch/job"
+```
+
+This will create a Kueue deployment that will manage batch jobs. 
+
+#### RHOAI Enablement
+
+Red Hat Openshift AI is already using Kueue in production. Their deployment can be replicated:
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ labels:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+  config:
+     integrations:
+      frameworks:
+       - "batch/job"
+       - "kubeflow.org/mpijob"
+       - "ray.io/rayjob"
+       - "ray.io/raycluster"
+       - "jobset.x-k8s.io/jobset"
+       - "kubeflow.org/mxjob"
+       - "kubeflow.org/paddlejob"
+       - "kubeflow.org/pytorchjob"
+       - "kubeflow.org/tfjob"
+       - "kubeflow.org/xgboostjob"
+      externalFrameworks:
+       - "AppWrapper.v1beta2.workload.codeflare.dev"
+```
+
+RHOAI provides developer features and allows for enablement of more advanced features. 
+Due to this, it is requested that Kueue Operator will still allow the
+changing of feature gates to provide more advanced functionality.
+
+### API Extensions
+
+I listed the API above.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+We want to support Hypershift. 
+This operator will allow for one to install in a namespace separate from a core openshift namespace.
```

Comment:
can you elaborate on this? Why does this need to be any different to standalone?

Reply:
I was told from PM that this is a requirement for hypershift.

Reply:
Hey! I met with a few folks on HCP to discuss this. I updated with a lot more information. Thank you for calling this out.